src/locks.rs

```diff
@@ -0,0 +1,146 @@
+//! Lock strategies.
+
+#[cfg(unix)]
+use std::os::fd::AsRawFd;
+use std::{
+    fs::File,
+    ops::{Deref, DerefMut},
+};
+
+use memmap2::MmapMut;
+
+use crate::synchronizer::SynchronizerError;
+
+/// The write lock strategy supports different lock implementations which can be chosen based on
+/// the guarantees required, platform support, and performance constraints.
+pub trait WriteLockStrategy<'a> {
+    type Guard: DerefMut<Target = MmapMut> + 'a;
+
+    /// Create a new instance of this lock strategy.
+    ///
+    /// The `mmap` parameter will have write access controlled by the lock.
+    ///
+    /// The `file` parameter is required because lock strategies depending on `flock` must hold
+    /// the file descriptor open so the kernel does not release the lock.
+    fn new(mmap: MmapMut, file: File) -> Self;
+
+    /// Provide read access to mmaped memory.
+    fn read(&'a self) -> &'a MmapMut;
+
+    /// Acquire the lock as specified by the lock strategy.
+    ///
+    /// On success, return a lock guard which can be used to access the underlying mmaped memory
+    /// via [`Deref`]/[`DerefMut`].
+    fn lock(&'a mut self) -> Result<Self::Guard, SynchronizerError>;
+}
+
+/// Lock protection is disabled.
+///
+/// # Safety
+/// Callers must ensure that there is only a single active writer. For example, the caller might
+/// ensure that only one process attempts to write to the synchronizer, and ensure that multiple
+/// instances of the process are not spawned.
+pub struct Disabled(MmapMut);
```

Comment:
Perhaps, name it `LockDisabled` for better clarity.

Reply:
Updated.